src/main/java/org/killbill/billing/plugin/analytics/api/user/AnalyticsUserApi.java

```diff
@@ -114,4 +117,35 @@ public void rebuildAnalyticsForAccount(final UUID accountId, final CallContext c
         allBusinessObjectsDao.update(businessContextFactory);
         logger.info("Finished Analytics refresh for account {}", businessContextFactory.getAccountId());
     }
+
+    public void rebuildAnalyticsForAllAccounts(final CallContext context) {
+        logger.info("Starting Analytics refresh for all accounts");
+        Pagination<Account> accounts = osgiKillbillAPI.getAccountUserApi().getAccounts(0L, 100L, context);
+        rebuildAnalyticsForAccounts(accounts, context);
+        Long nextOffSet = accounts.getNextOffset();
+        while (nextOffSet != null) {
+            accounts = osgiKillbillAPI.getAccountUserApi().getAccounts(nextOffSet, 100L, context);
+            rebuildAnalyticsForAccounts(accounts, context);
+            nextOffSet = accounts.getNextOffset();
+        }
+
+        logger.info("Finished Analytics refresh for all accounts");
+    }
+
+    private void rebuildAnalyticsForAccounts(final Pagination<Account> accounts, final CallContext context) {
+        final Iterator<Account> accountsItr = accounts.iterator();
+        while (accountsItr.hasNext()) {
```

Comment:
Fixed. I've handled the `IOException` within the `finally` block itself. Let me know if you think we should instead throw the `IOException`out of this method and handle it in the `AnalyticsAccountsResource` by returning an error status code to the client.

Reply:
Maybe you could return a small JSON object which tells the user how many refreshes were successfully scheduled? Something like:

```
{
  "nbAccounts": 120,
  "nbRefreshes": 112
}
```
In the above, this would mean 8 errors.

BTW were you able to test it when your Kill Bill installation has more than 100 accounts, just to make sure the pagination logic works well?

Reply:
Fixed, tested with my local Kill Bill installation that has around 240 records, seems to work fine.